lib/filewatch/watch.rb

```diff
@@ -118,13 +118,21 @@ def each(&block)
       synchronized do
         return if @files.empty?
 
+        quitting = false
         file_deleteable = []
         # creates this array just once
         watched_files = @files.values
 
+        quit_proc = lambda do
```

Comment:
yeah, this would require a move of the synchronised call to outside of the each method..so instead of 

```ruby
def each(&block)
  synchronised do
    # ...
  end
end
```

instead do

```ruby
# ...
   synchronised do
     each(&block)
   end
```